php_phongo.c

```diff
@@ -1561,11 +1553,6 @@ static void php_phongo_cursor_iterator_dtor(zend_object_iterator *iter TSRMLS_DC
 {
 	php_phongo_cursor_iterator *cursor_it = (php_phongo_cursor_iterator *)iter;
 
-	if (cursor_it->cursor->firstBatch) {
-		bson_clear(&cursor_it->cursor->firstBatch);
-		cursor_it->cursor->firstBatch = NULL;
-	}
-
 	php_phongo_cursor_free_current(cursor_it->cursor);
```

Comment:
But the "free_object" handler already calls `php_phongo_cursor_free_current()`. So it would either implicitly be invoked by us dtor-ing the zval here (if it's the last reference) or later on (if there is some other reference). And if there was another reference, there's probably no need to free the current element. If a second iterator is created for the Cursor (i.e. `php_phongo_cursor_t` struct), then its own rewind would free the current element anyway, but that's a side effect of two PHP iterators sharing the same Cursor.